### PR TITLE
Click cooldown refactor

### PR DESCRIPTION
**Pull request by PowerfulBacon**
## About The Pull Request

Refactors click cooldowns to have a lot more versatility. This makes it significantly easier to increase the action cooldown during the attack chain since you don't have to figure out a way to apply it after the initial cooldown is applied.
This also allows you to add cooldowns which won't be overriden by certain sources, for example mech attacks won't be sped up by a mob having a higher action speed modifier for their user actions.

## Why It's Good For The Game

See above. Needed for other code projects I am working on.

## Testing Photographs and Procedure

Tested regular attacking, toy guardians being fast and gloves of the north star being fast.

See unit test results for proof of all of the functions working.

image

https://github.com/BeeStation/BeeStation-Hornet/assets/26465327/1840396a-35b5-4a0b-923f-821d1c8ee833

## Changelog
:cl:
code: Click cooldown refactoring.
/:cl:

### CODE REVIEW

File: code/__DEFINES/cooldown.dm [hunk 1]
```diff
@@ -0,0 +1,23 @@
+
+// Action cooldown groups. Starts at 1
+#define CD_GROUP_USER_ACTION 1	// Actions which apply to the user
+#define CD_GROUP_TELEKENISIS 2	// Telekenetic actions
```
Comment (itsmeow) [inline code/__DEFINES/cooldown.dm]:
```suggestion
#define CD_GROUP_TELEKINESIS 2	// Telekinetic actions
```

**Review by Rukofamicom — CHANGES_REQUESTED** [review]
Fix merge conflict and commit Kat's change, then will merge.

**Comment by PowerfulBacon** [comment]
Conflicts resolved, worked in testing. Wasn't able to properly test holoparasites due to requiring 2 people and having no easilly accessible debugging tools that I could find in VV

**Review by Rukofamicom — APPROVED** [review]
I'm sure we'll hear about it soon if holopara breaks lol

**Comment by Rukofamicom** [comment]
Ah rip lol

**Comment by PowerfulBacon** [comment]
This will need a proper deconflict since it moved a CD cooldown to another place.

**Comment by itsmeow** [comment]
```
code/game/turfs/turf_integrity.dm, line 190, column 6:
error: undefined proc: "changeNext_move" on /mob/living

code/game/turfs/turf_integrity.dm, line 278, column 6:
error: undefined proc: "changeNext_move" on /mob
```

**Comment by Rukofamicom** [comment]
coming up on two months conflicted now

**Pull request closed without merging**